Add HiFive1 FE310-based board support

kgugala opened the pull request:
This adds support for the SiFive's HiFive1 board with FE310 SPI controller support.
Closes issue #3650

galak commented on boards/riscv32/hifive1/clock.c:
```diff
@@ -0,0 +1,29 @@
+
+	config_value |= PLL_REFSEL(1);
+	config_value |= PLL_BYPASS(1);
+	PRCI_REG(PRCI_PLLCFG) = config_value;
```
Any reason to not just have:

PRCI_REG(PRCI_PLLCFG) = PLL_REFSEL(1) | PLL_BYPASS(1);

galak commented on boards/riscv32/hifive1/pinmux.c:
```diff
@@ -0,0 +1,35 @@
+	pinmux_pin_set(p, 17, FE310_PINMUX_IOF0);
+
+	/* SPI1 */
+
```
nit: remove blank line

galak reviewed (changes requested):
Some minor nits on the board code.

galak commented:
Can you pull the SPI driver out into a separate PR.  It should probably use the new API.  See PR #5921

galak reviewed (changes requested):
You need board docs for a new board.

kgugala commented:
Hi @galak, thanks for the review and hints. I agree that the SPI should be a separate pull request. Once #5921 is merged I will create new PR for the SPI driver.

I have added the board docs, and fixed the issues you pointed. Please review the code again.

galak reviewed (changes requested):
merge the board doc commit with the commit that adds the board.

galak commented:
> done, doc adding commit is merged with the one adding board

Thanks, hopefully @dbkinder has a few minutes to review the docs.

dbkinder commented on boards/riscv32/hifive1/doc/hifive1.rst:
```diff
@@ -0,0 +1,60 @@
+Overview
+********
+
+The HiFive1 is a development board with the FE310 RISC-V SoC.
```
I'd merge these first two sentences:

    The HiFive1 is an Arduino-compatible development board with an FE310 RISC-V SoC.

dbkinder commented on boards/riscv32/hifive1/doc/hifive1.rst:
```diff
@@ -0,0 +1,60 @@
+
+.. code-block:: bash
+
+	$ export ZEPHYR_GCC_VARIANT="zephyr"
```
We're trying not to include the ``$`` prompt to make it easier to cut and paste commands.  Also, don't indent so far (use spaces rather than a tab):

    .. code-block:: bash

       export ZEPHYR_GCC_VARIANT="zephyr"
       export ZEPHYR_SDK_INSTALL_DIR="/opt/zephyr-sdk" #assuming default sdk install directory
       export BOARD="hifive1"

Actually, are the first two ``export`` commands needed?

dbkinder commented on boards/riscv32/hifive1/doc/hifive1.rst:
```diff
@@ -0,0 +1,60 @@
+========
+
+In order to upload the application to the device, you'll need OpenOCD and GDB with RISC-V support.
+Those are easiest obtainable from SiFive's Freedom-E-SDK. Download and installation instructions can be found in the `SDK's GitHub repository <https://github.com/sifive/freedom-e-sdk>`_.
```
Replace this whole line with:

    Download and installation instructions can be found in the `SiFive's Freedom-E-SDK
    GitHub repository <https://github.com/sifive/freedom-e-sdk>`_.

dbkinder commented on boards/riscv32/hifive1/doc/hifive1.rst:
```diff
@@ -0,0 +1,60 @@
+In order to upload the application to the device, you'll need OpenOCD and GDB with RISC-V support.
+Those are easiest obtainable from SiFive's Freedom-E-SDK. Download and installation instructions can be found in the `SDK's GitHub repository <https://github.com/sifive/freedom-e-sdk>`_.
+
+With the necessary tools installed, you can connect to the board using OpenOCD. To establish an OpenOCD connection, switch to the :file:`utils` directory and run:
```
Please keep line length to < 80 characters

dbkinder commented on boards/riscv32/hifive1/doc/hifive1.rst:
```diff
@@ -0,0 +1,60 @@
+
+.. code-block:: bash
+
+	# assuming that the location of the openocd is in PATH
```
align the code block contents to the ``c`` in ``code-block`` (and leave off the ``$`` prompt)

dbkinder commented on boards/riscv32/hifive1/doc/hifive1.rst:
```diff
@@ -0,0 +1,60 @@
+	$ sudo openocd -f ${SDK_PATH}/bsp/env/freedom-e300-hifive1/openocd.cfg
+
+
+Leave it running, and in a different terminal, use GDB to upload the binary to the board. Use the RISC-V GDB from SiFive's Freedom-E-SDK toolchain.
```
< 80 characters per line please

dbkinder commented on boards/riscv32/hifive1/doc/hifive1.rst:
```diff
@@ -0,0 +1,60 @@
+
+Leave it running, and in a different terminal, use GDB to upload the binary to the board. Use the RISC-V GDB from SiFive's Freedom-E-SDK toolchain.
+
+Before loading, the device's flash protection has to be disabled. In order to load the binary to the device, run the following commands in the GDB terminal:
```
Change to:

    Before loading the binary image to the device, the device's flash protection
    must be disabled. Here are the GDB terminal commands to connect to the device,
    disable the flash protection, load the binary, and start it running:

dbkinder commented on boards/riscv32/hifive1/doc/hifive1.rst:
```diff
@@ -0,0 +1,60 @@
+
+.. code-block:: console
+
+	$ gdb
```
Align the commands with the ``c`` in ``code-block``

dbkinder commented:
Sorry for the delay.  I was out of the office the last week.

Commit:
arch: riscv32: fe310: Always-On domain adress definition

Added always-on domain address definition

Signed-off-by: Tomasz Michalak <tmichalak@antmicro.com>
Signed-off-by: Karol Gugala <kgugala@antmicro.com>
Signed-off-by: Tomasz Jurtsch <tjurtsch@antmicro.com>

Commit:
boards: SiFive HiFive1 board

Added support for the HiFive1 board from SiFive

Signed-off-by: Tomasz Michalak <tmichalak@antmicro.com>
Signed-off-by: Karol Gugala <kgugala@antmicro.com>
Signed-off-by: Tomasz Jurtsch <tjurtsch@antmicro.com>

kgugala commented:
Thanks @dbkinder. All the requested changes are now included in the docs.

dbkinder reviewed (approved):
Thanks!
